### PR TITLE
client-side POC

### PR DESCRIPTION
**Pull request by ro0NL**
for reference https://github.com/beyondcode/laravel-view-xray/blob/master/resources/assets/xray.js

### CODE REVIEW

**Comment by ro0NL** [comment]
image

**Comment by ro0NL** [comment]
image

**Comment by anyt** [comment]
Does it outline "Included template"?

**Comment by ro0NL** [comment]
yes, that is `inc.html.twig`. But note ive added a block element below it

https://jsfiddle.net/h9gs21Lt/3/

File: templates/client.js.twig [hunk 1]
```diff
@@ -0,0 +1,110 @@
+var js = (function() {
```
Comment (stof) [inline templates/client.js.twig:2]: you should avoid leaking a global variable named `js`. Thus, you only use it just after creating it, so everything could go inside the closure.

Comment (ro0NL) [reply templates/client.js.twig:2]: @stof this is only some boilerplate to replicate the Webprofilerbundle, the `Sfjs` variable: https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig#L8

Comment (stof) [reply templates/client.js.twig:2]: but here, you don't actually need to leak a global variable.

the profiler needs to create it, to expose its JS to custom panels.

Comment (yceruto) [reply templates/client.js.twig:2]: I agree with @stof here "the global variable should be avoided" if it isn't completely necessary.

**Comment by yceruto** [comment]
LGTM, can we remove the hash and find a better format? I'm thinking something like
```
- "<template-name>" line <n>
- Block "name" in "<template-name>" line <n>
- Macro "name" in "<template-name>" line <n>
```
Later in WPB could we add a link option to open the file#line?

**Comment by ro0NL** [comment]
the hash is at least needed to match the closing boundary.

in general i'd like the tool to be super simple and scoped to HTML only (hence i suggest to rename it to `HtmlDebugNodeVisitor`) Overall: some visual element > click > source file.

In WPB it would use the file link helper to get us the file link, which we need to render server side :thinking: 

Also profiler templates should not be affected (e.g. the WDT itself). Ideally server-side we dont render comments then... but this can be solved at the client level also.

**Comment by yceruto** [comment]
> the hash is at least needed to match the closing boundary.

yes, I meant the tooltip representation.

**Comment by ro0NL** [comment]
ah :) suggestions?

**Comment by yceruto** [comment]
let me know if this is ready for you.

**Comment by ro0NL** [comment]
ready so far :) a type TEMPLATE will always be line 1 no? i left it out in this case.

**Review by yceruto — APPROVED** [review]

**Pull request merged**